File: practise/leetcode/DailyQuestion/202410/1436旅行的终点站.ts
function destCity(paths: string[][]): string {
    const sa = new Set();
    for (const p of paths){
        sa.add(p[0]);
    }
    for (const p of paths){
        if(!sa.has(p[1])){
            return p[1];
        }
    }
    return "";
};

File: practise/leetcode/DailyQuestion/202410/1436旅行的终点站.cpp
class Solution {
public:
    string destCity(vector<vector<string>>& paths) {
        unordered_set<string> sa, sb;
        for(auto& p : paths){
            auto& a = p[0];
            auto& b = p[1];
            sb.erase(a);
            if(!sa.contains(b)){
                sb.insert(b);
            }
            sa.insert(a);
        }

    return *sb.begin();
    }
};